### PR TITLE
Dev: Add Visual Studio Code + Github Codespaces Support

### PR DESCRIPTION
**Pull request by JustinGrote**
This adds: 

- [x] Support for [ Visual Studio Code Docker Remote Development](https://code.visualstudio.com/docs/remote/containers). 
- [x] Support for [Github Codespaces](https://github.com/features/codespaces/) once it becomes available for the repo.
- [x] Default debug task (F5) using Bash Debug so that developers can step through the linter script for debugging and validation

Capture

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+	// Sets the run context to one level up instead of the .devcontainer folder.
+	"context": "..",
+
+	// Update the 'dockerFile' property if you aren't using the standard 'Dockerfile' filename.
+	"dockerFile": "..\\Dockerfile",
+
+	// Set *default* container specific settings.json values on container create.
+	"settings": { 
+		"terminal.integrated.shell.linux": "/bin/ash"
```
Comment (antleblanc) [inline .devcontainer/devcontainer.json:14]:
```suggestion
		"terminal.integrated.shell.linux": "/bin/bash"
```

Comment (JustinGrote) [reply .devcontainer/devcontainer.json:14]: @antleblanc the default terminal in Alpine is Ash and that was being used as the base, I'm OK with this alteration though.

**Comment by zkoppert** [comment]
@JustinGrote This looks really awesome! 🎉 I don't personally use VSCode but would like to try this out. Would you be willing to write some documentation on how to get started with this? Maybe a good place to put that would be in the `README.md` under the debugging section?

**Comment by JustinGrote** [comment]
@zkoppert It's already covered by the vscode documentation, you basically just choose "open in container" and the rest is done automatically per the devcontainer.json file.

**Comment by zkoppert** [comment]
Good point. Maybe just an advertisement then in the debugging section of the readme saying "You can utilize the debug feature of VSCode as this project already contains the needed configuration"? I want to make sure that debugging is accessible to new contributors. Maybe include your GIF there to help lead people to understanding?

**Comment by JustinGrote** [comment]
> Good point. Maybe just an advertisement then in the debugging section of the readme saying "You can utilize the debug feature of VSCode as this project already contains the needed configuration"? I want to make sure that debugging is accessible to new contributors. Maybe include your GIF there to help lead people to understanding?

Added.

**Pull request merged**